pkg/imgpkg/cmd/imgpkg.go

```diff
@@ -64,6 +66,21 @@ func NewImgpkgCmd(o *ImgpkgOptions) *cobra.Command {
 
 	cobrautil.VisitCommands(cmd, cobrautil.WrapRunEForCmd(cobrautil.ResolveFlagsForCmd))
 
+	cobrautil.VisitCommands(cmd, cobrautil.WrapRunEForCmd(func(cmd *cobra.Command, args []string) error {
+		protocolMatcher := regexp.MustCompile(`^https*://`)
+
+		bundleFlag := cmd.Flag("bundle")
+		if bundleFlag != nil && protocolMatcher.MatchString(bundleFlag.Value.String()) {
+			return fmt.Errorf("bundle flag %v starts with protocol: remove http(s):// from bundle", bundleFlag.Value)
+		}
+
+		imageFlag := cmd.Flag("image")
+		if imageFlag != nil && protocolMatcher.MatchString(imageFlag.Value.String()) {
+			return fmt.Errorf("image flag %v starts with protocol: remove http(s):// from image", imageFlag.Value)
+		}
+		return nil
```

Comment:
not sure if there's a third example of bundle/image flag that we should check,  if so obvs should extract the pattern to one more func and just call it for each flag. 

also could get fancier w. regexps / protocols ,  e.g. could match `^.{1,10}://` to catch any protocol in case somebody has their bundle sitting on a gopher server or something.  (e.g. FTP seems vaguely possible).   If we did this then seems reasonable to use a capture group or something to show exactly what was matched.